toolchain/check/import_cpp.cpp

```diff
@@ -146,32 +146,23 @@ class CarbonClangDriverDiagnosticConsumer : public clang::DiagnosticConsumer {
 
 }  // namespace
 
-// Builds a clang `CompilerInvocation` describing the options to use to build an
-// imported C++ AST.
-// TODO: Cache the compiler invocation created here and reuse it if building
-// multiple AST units. Consider building the `CompilerInvocation` from the
-// driver and passing it into check. This would also allow us to have a shared
-// set of defaults between the Clang invocation we use for imports and the
-// invocation we use for `carbon clang`.
-static auto BuildCompilerInvocation(
-    Context& context, llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem> fs,
-    const std::string& clang_path, const std::string& target)
+auto BuildClangInvocationImpl(
+    Diagnostics::NoLocEmitter& emitter,
+    llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem> fs,
+    llvm::ArrayRef<std::string> clang_path_and_args)
```

Comment:
It seems pretty awkward to leave this in `import_cpp.cpp`, and in `check` really, especially as there is only *one* invocation used for a given compile...

AFAICT, there aren't many dependencies on being here unless I'm missing something? (Sorry if so and I just missed it...). Given that, WDYT about moving this into a separate component that both the driver and the language server can use? I'm thinking of a library in `base` like `clang_interop_helpers.h` or something similar? That would then make sense to expose both an API to build an interop-oriented compiler invocation and just get the interop-oriented driver arguments (long term)?

I think that would also nudge us towards some improvements such as using a more generic name than `IncludesFileName`, maybe `InteropFileName`?

Reply:
Yeah, the only reason I left this here was to avoid exposing the file name, which really feels like a (very ugly) implementation detail. Other than that, I agree it would fit more nicely somewhere else. Will look into that.

I wonder if we can have the code that runs the compiler either read the filename out of the invocation or replace it, so that we don't even need to rely on it being the same string.

Reply:
Done.